src/code/graph.c

```diff
@@ -0,0 +1,527 @@
+
+    osViSwapBuffer(newFb);
+    Fault_WaitForInput();
+    osViSwapBuffer(nextFb);
+}
+
+void Graph_DisassembleUCode(void* arg0) {
+    UcodeDisas disassembler;
+
+    if (HREG(80) == 7 && HREG(81) != 0) {
+        func_800D7F5C(&disassembler);
+        disassembler.enableLog = HREG(83);
+        func_800DAC80(&disassembler, 3, D_8012D230);
+        func_800DAC90(&disassembler, D_80155F50);
+        func_800D8400(&disassembler, arg0);
+        HREG(93) = disassembler.dl_cnt;
+        HREG(84) =
+            disassembler.tri2_cnt * 2 + disassembler.tri1_cnt + (disassembler.quad_cnt * 2) + disassembler.line_cnt;
+        HREG(85) = disassembler.vtx_cnt;
+        HREG(86) = disassembler.spvtx_cnt;
+        HREG(87) = disassembler.tri1_cnt;
+        HREG(88) = disassembler.tri2_cnt;
+        HREG(89) = disassembler.quad_cnt;
+        HREG(90) = disassembler.line_cnt;
+        HREG(91) = disassembler.sync_err;
+        HREG(92) = disassembler.loaducode_cnt;
+        if (HREG(82) == 1 || HREG(82) == 2) {
+            osSyncPrintf("vtx_cnt=%d\n", disassembler.vtx_cnt);
+            osSyncPrintf("spvtx_cnt=%d\n", disassembler.spvtx_cnt);
+            osSyncPrintf("tri1_cnt=%d\n", disassembler.tri1_cnt);
+            osSyncPrintf("tri2_cnt=%d\n", disassembler.tri2_cnt);
+            osSyncPrintf("quad_cnt=%d\n", disassembler.quad_cnt);
+            osSyncPrintf("line_cnt=%d\n", disassembler.line_cnt);
+            osSyncPrintf("sync_err=%d\n", disassembler.sync_err);
+            osSyncPrintf("loaducode_cnt=%d\n", disassembler.loaducode_cnt);
+            osSyncPrintf("dl_depth=%d\n", disassembler.dl_depth);
+            osSyncPrintf("dl_cnt=%d\n", disassembler.dl_cnt);
+        }
+        func_800D7FC4(&disassembler);
+    }
+}
+
+void Graph_UCodeFaultClient(void* arg0) {
+    UcodeDisas disassembler;
+
+    func_800D7F5C(&disassembler);
+    disassembler.enableLog = true;
+    func_800DAC80(&disassembler, 3, D_8012D248);
+    func_800DAC90(&disassembler, D_80155F50);
+    func_800D8400(&disassembler, arg0);
+    func_800D7FC4(&disassembler);
+}
+
+void* Graph_InitTHGA(GraphicsContext* gfxCtx) {
+    GfxPool* pool;
+
+    pool = &gGfxPools[gfxCtx->gfxPoolIdx & 1];
+
+    pool->headMagic = GFXPOOL_HEAD_MAGIC;
+    pool->tailMagic = GFXPOOL_TAIL_MAGIC;
+    THGA_Ct(&gfxCtx->polyOpa, pool->polyOpaBuffer, sizeof(pool->polyOpaBuffer));
+    THGA_Ct(&gfxCtx->polyXlu, pool->polyXluBuffer, sizeof(pool->polyXluBuffer));
+    THGA_Ct(&gfxCtx->overlay, pool->overlayBuffer, sizeof(pool->overlayBuffer));
+    THGA_Ct(&gfxCtx->work, pool->workBuffer, sizeof(pool->workBuffer));
+
+    gfxCtx->polyOpaBuffer = pool->polyOpaBuffer;
+    gfxCtx->polyXluBuffer = pool->polyXluBuffer;
+    gfxCtx->overlayBuffer = pool->overlayBuffer;
+    gfxCtx->workBuffer = pool->workBuffer;
+
+    gfxCtx->curFrameBuffer = (u16*)SysCfb_GetFbPtr(gfxCtx->fbIdx % 2);
+    gfxCtx->unk_014 = 0;
+}
+
+GameStateOverlay* Graph_GetNextGameState() {
+    void* game_init_func;
```

Comment:
This name needs to be changed to fit our naming scheme and the fact that we use "game state" instead of just "game".
```suggestion
    void* gameStateInitFunc;
```